File: Array/66. Plus One.py
class Solution:
    def temp(self, digits: List[int], index: int):
        if(digits[index]<10):
            return digits[index]
        elif index>=len(digits)-1:
            digits.append(1)
        else:
            digits[index+1] = digits[index+1] + 1
        return digits[index]-10
    def plusOne(self, digits: List[int]) -> List[int]:
        digits.reverse()
        digits[0] = digits[0]+1
        digits = [self.temp(digits,i) for i, num in enumerate(digits)]
        digits.reverse()
        return digits

File: Array/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int add=1;
        int index = digits.size()-1;
        
        
        if(digits[index]<9){
            digits[index]++;
            return digits;
        }
        int i=0;
        while(i<=index&&digits[i]==9){
            i++;
        }
        if(i>index){
            digits[0]=1;
            for(int j=1;j<=index;j++){
                digits[j]=0;
            }
            digits.push_back(0);
            return digits;
        }
        while(add==1){
            if(digits[index]==9){
                digits[index]=0;
                index--;
            }else{
                digits[index]++;
                add=0;
            }
        }
        return digits;
        
    }
};